File: 41to60/58_Length_of_Last_Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        length = 0
        p1 = 0

        while(p1 < len(s)):
            print(s[p1],length)
            # if the p1 is not a space
            if(s[p1] != ' '):
                length += 1
            # ignore everything by reseting the length if theres more to the string
            elif(p1+1 < len(s) and s[p1+1] != ' '):
                length = 0

            p1+= 1 

        return length

File: 41to60/58_Length_of_Last_Word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int p1 = 0;
        int n = 0;

        while(p1 < s.length()){
            if(s[p1] != ' '){
                n++;
            }
            else if(p1+1 < s.length() && s[p1+1] != ' '){
                n = 0;
            }
            p1++;
        }
        return n;
        
    }
};